### ISSUE TITLE
Disable "SetTensor/CopyTensor" console logging.

### ISSUE THREAD

**Issue by ZappaBoy** [ROCmSoftwarePlatform/MIOpen#2247]
### Issue description
I cannot be able to disable some console logging about `SetTensor/CopyTensor`. This is quite annoying because the logs are printed constantly  surrounding other logs. 

Logging example:
```shell
...
SetTensor
real descritor: 1, 1, 64
flat descritor: 64
SetTensor
real descritor: 1, 1, 64
flat descritor: 64
CopyTensor
src real descriptor: 1, 1, 64
src flat descriptor: 64
dst real descriptor: 1, 1, 64
dst flat descriptor: 64
...
```

I also tried to disable logging using the `MIOPEN_ENABLE_LOGGING`, `MIOPEN_ENABLE_LOGGING_CMD`, `MIOPEN_LOG_LEVEL`, and `TF_CPP_MAX_VLOG_LEVEL` without success. This is an example in `python`, I also tried setting directly in mine `/etc/profile` or using `export` command.

```python
self.rocm_logging_level = -1  # -1 disable logging
os.environ['MIOPEN_ENABLE_LOGGING'] = f'{1 if rocm_logging_level > 0 else 0}'
os.environ['MIOPEN_ENABLE_LOGGING_CMD'] = f'{1 if rocm_logging_level > 0 else 0}'
os.environ['MIOPEN_LOG_LEVEL'] = f'{rocm_logging_level}'
os.environ['TF_CPP_MAX_VLOG_LEVEL'] = f'{rocm_logging_level}'
os.environ['HSAKMT_DEBUG_LEVEL'] = f'{rocm_logging_level}'
```

Here are the values of the environment variables above at running time:
```python
MIOPEN_ENABLE_LOGGING: 0
MIOPEN_ENABLE_LOGGING_CMD: 0
MIOPEN_LOG_LEVEL: 0
TF_CPP_MAX_VLOG_LEVEL: 0
HSAKMT_DEBUG_LEVEL: 0
```

I found the where the logs are printed in `MIOpen` code: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/cec8c587c1e6e676dbf42a58f96c3fed27828cf6/src/ocl/tensorocl.cpp#L1736-L1745

But I do not understand why only these logs keep showing while the other logs gone.

I also found a probably related issue https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1057 but there is no clear explanation on how to fix this.

### Possible fix
Is it possible that the problem is the `#ifndef` directive on `NDEBUG` macro? Maybe simply use the `#ifdef` directive can fix that?

### OS Platform, distribution and device
ArchLinux 6.1.38-1-lts - AMD Radeon RX 6700 XT - gfx1031

**Comment by junliume** [comment]
@JehandadKhan @DrizztDoUrden @averinevg :
Could you check if this issue has a quick WA? It does impact end user experience and I am curious why these info messages get printed to stdout too.

**Comment by DrizztDoUrden** [comment]
TL; DR: It looks like you have a debug build of MIOpen. In debug MIOPEN_LOG_LEVEL defaults to 5. If any MIOpen call happens before the env value changes, the value (even empty) gets cached.

Details:

`MIOPEN_LOG_I` prints only when `MIOPEN_LOG_LEVEL` is at least 5 https://github.com/ROCmSoftwarePlatform/MIOpen/blob/3f206736b981adb915a29f8030b73adb7a6f71b5/src/include/miopen/logger.hpp#L194
Are you sure you change env variables before the first MIOpen call? If there is any MIOpen call before changing the environment, previous values are cached. Even tensor/handle initialization calls may cache these values for the whole library. Or maybe your changes are not actually applied to the process calling MIOpen.

> Is it possible that the problem is the #ifndef directive on NDEBUG macro? Maybe simply use the #ifdef directive can fix that?

NDEBUG should be defined only in release builds. So `#ifndef NDEBUG` means `if NDEBUG is not defined`, which is like `if not release`. So this is likely a debug build of the library, or some release build with manually added NDEBUG. In that case the default warning level is 5 (info). So, if the previous (possibly, empty) value is cached, 5 may be used for all future calls.

> Could you check if this issue has a quick WA? It does impact end user experience and I am curious why these info messages get printed to stdout too.

Technical corrections, MIOPEN_LOG... macros print to stderr, rather than to stdout.

**Comment by ZappaBoy** [comment]
First of all,  thank you all for your time spent on this issue and in general on the whole `ROCm` project.


> TL; DR: It looks like you have a debug build of MIOpen. In debug MIOPEN_LOG_LEVEL defaults to 5. If any MIOpen call happens before the env value changes, the value (even empty) gets cached.
> 
> Details:
> 
> `MIOPEN_LOG_I` prints only when `MIOPEN_LOG_LEVEL` is at least 5
> 
> https://github.com/ROCmSoftwarePlatform/MIOpen/blob/3f206736b981adb915a29f8030b73adb7a6f71b5/src/include/miopen/logger.hpp#L194
> 
> 
> Are you sure nothing changes the env variables before MIOpen calls? If there is any MIOpen call before changing the environment, previous values are cached. Even tensor/handle initialization calls may cache these values for the whole library.
> > Is it possible that the problem is the #ifndef directive on NDEBUG macro? Maybe simply use the #ifdef directive can fix that?
> 
> NDEBUG should be defined only in release builds. So `#ifndef NDEBUG` means `if NDEBUG is not defined`, which is like `if not release`. So this is likely a debug build of the library, or some release build with manually added NDEBUG. In that case the default warning level is 5 (info). So, if the previous (possibly, empty) value is cached, 5 may be used for all future calls.
> 
> > Could you check if this issue has a quick WA? It does impact end user experience and I am curious why these info messages get printed to stdout too.
> 
> Technical corrections, MIOPEN_LOG... macros print to stderr, rather than to stdout.

Before opening this issue I tried to set globally these envs on startup setting them directly in my `/etc/profile`. Also, I set again the envs value in the code to ensure the different logging levels on the project. 
Now, I set again the `MIOPEN_LOG_LEVEL` env to `3` in the `/etc/profile` and nothing changed (obviously I rebooted). 
I also set it manually in the code to the same value as my globally exported variable (`MIOPEN_LOG_LEVEL=3`) and the result is the same. 
Also, I tried to set both the `MIOPEN_ENABLE_LOGGING` and `MIOPEN_ENABLE_LOGGING_CMD` to `0` in order to completely disable the logs but it seems like all the logging envs are ignored. 

Is there a way to check if the envs are cached or simply clean the cache before setting the logging envs?

Just so your know, I'm currently on Arch Linux distro so I installed the `ROCm` packages using the `AUR` packages from [rocm-arch](https://github.com/rocm-arch/rocm-arch) without directly building it. Honestly, I don't think that the `rocm-arch` are providing a debug build but just to be sure I'll check that deeper.

**Comment by ZappaBoy** [comment]
> Just so your know, I'm currently on Arch Linux distro so I installed the `ROCm` packages using the `AUR` packages from [rocm-arch](https://github.com/rocm-arch/rocm-arch) without directly building it. Honestly, I don't think that the `rocm-arch` are providing a debug build but just to be sure I'll check that deeper.

Never mind, the problem is related to the `rocm-build` I found an open issue about it (https://github.com/rocm-arch/rocm-arch/issues/952). 

Thank you all for your help and forgive me for wasting your time.

**Comment by ZappaBoy** [comment]
FYI right today (perfect timing) this issue has been solved due to the version `5.6` of `rocm` has been released on `rocm-arch` on the Arch Linux `extra` repository. 
I'm going to close this issue. Thanks again.
